Scenes/ContentManager/Custom_Editors/TacticalMapEditor/Scripts/TacticalMapEditorTileBrush.gd

```diff
@@ -23,3 +34,30 @@ func set_selected(is_selected: bool) -> void:
 		modulate = Color(0.227, 0.635, 0.757)
 	else:
 		modulate = Color(1,1,1)
+
+
+# Set the label text and adjust the size of the label and the tile sprite accordingly
+func set_label(text: String):
+	label.text = text
+	label.custom_minimum_size = Vector2(DEFAULT_WIDTH, 12)
+
+	label.autowrap_mode = TextServer.AUTOWRAP_WORD_SMART  # Enable word wrapping
+	label.size_flags_horizontal = Control.SIZE_FILL
+	label.size_flags_vertical = Control.SIZE_SHRINK_END
+	label.custom_minimum_size = Vector2(DEFAULT_WIDTH, label.get_minimum_size().y)
+
+	# Update the minimum size of the parent container
+	update_min_size()
+
+
+func _on_label_minimum_size_changed():
+	# Update the minimum size of the parent container
+	update_min_size()
+
+
+# Update the minimum size of the parent container
+func update_min_size():
+	var total_height = DEFAULT_WIDTH + label.custom_minimum_size.y
+	# Because the text might wrap around to the next line, we add 20 height to accommodate
+	# that. If the label text does not wrap around, the 20 height will be empty space
+	custom_minimum_size = Vector2(custom_minimum_size.x, total_height+20)
```

Comment:
Can we also add a const for this `20`? Right now it doesn't describe what it really is and it's confusing a little

Reply:
Added a const for line height